ui/src/app/machines/views/MachineList/MachineListHeader/ActionFormWrapper/CommissionForm/CommissionFormFields/CommissionFormFields.js

```diff
@@ -16,6 +18,25 @@ export const CommissionFormFields = ({
     Object.keys(script.parameters).some((key) => key === "url")
   );
 
+  useEffect(() => {
+    setFieldValue("commissioningScripts", preselectedCommissioning);
+    setFieldValue("testingScripts", preselectedTesting);
+  }, [preselectedCommissioning, preselectedTesting, setFieldValue]);
+
+  useEffect(() => {
+    setDisabledScripts(
+      commissioningScripts.filter((script) => script.default === true)
+    );
+  }, [commissioningScripts]);
+
+  const [
+    showCommissioningPlaceholder,
+    setShowCommissioningPlaceholder,
+  ] = useState(true);
+
+  const [showTestingPlaceholder, setShowTestingPlaceholder] = useState(true);
+  const [disabledScripts, setDisabledScripts] = useState([]);
```

Comment:
Instead of a state variable and effect, could this just be determined in the `disabledTags` prop? i.e.:
`disabledTags={commissioningScripts.filter((script) => script.default === true)}`

Reply:
Was trying to avoid refiltering for every render but the cost is probably minimal.